### DISCUSSION TITLE
The z-axis and camera?

### DISCUSSION THREAD

**Discussion by Leedehai** [cetz-package/cetz#439] [Q&A] [answered]
Hi - thanks for making this powerful package! Since CeTZ supports XYZ coordinates, I'm wondering, by default,
* where does the z-axis point,
* what the camera model is, and
* what the camera position and angle is? (the lookAt matrix)

From the plots in the manual it appears the z-axis points inward the page & away from the reader, the camera uses orthographic projection, and the camera points from a point in (x>0,y>0,z>0) (but where, exactly?) to (0,0,0).

By the way, it'd be wonderful if this could be mentioned somewhere in the doc

Thanks😊

**Comment by johannes-wolf** [comment, marked as answer]
3D support is not really there as of now.

The default transformation just sets up z to be `(x=.5, y=-.5)` (which is a bit random...) (see canvas.typ line ~49) -- it set's up a z-shear matrix. So positive z points to positive x and negative y, which by default is to the top-right or "away" from the "camera". I guess TikZ does something very similar when setting `z=...`.

The 0.2.0 marks branch (+ path decorations) uses a `z-up` and `xy-up` vector style key to know where the up-vector should be. There are currently no functions to set up a projection matrix or any kind of "camera", although I would like to add that.

**Reply by Dictat111** [reply]
<img width="1542" height="693" alt="image" src="https://github.com/user-attachments/assets/bb44ecf9-9517-44c3-b77a-351b456d0582" />
How should I set the x, y, z?

**Reply by johannes-wolf** [reply]
What do you mean?

**Comment by michaelfortunato** [comment]
Are there further plans to make more abstractions for 3D diagrams? For now I am using the ortho environment.
